### PR TITLE
[kube-prometheus-stack] add support for enableRemoteWriteReceiver in the prometheuses crd and…

### PR DESCRIPTION
**Pull request by daveoy**
… write that out into our crd if its in our values file

Signed-off-by: Dave Young <davey@themill.com>

#### What this PR does / why we need it:

the prometheus operator supposedly supports this field in prometheusSpec, but we aren't passing it to the prometheus object via the kube-prometheus-stack helm chart, its also notably missing in the crd.

https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#prometheusspec

### Which issue this PR fixes
- fixes 1961

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
The CRDs will ve updated together with the prometheus-operator version, by a scrtipt: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/hack/update_crds.sh 

So please always use it and also update prometheus operator to 0.55.1, updating all docker iamges, dadhboards and rules too.

**Comment by HaroonSaid** [comment]
We are looking forward to turning on `enableRemoteWriteReceiver` 
Can it get merged?

**Comment by daveoy** [comment]
> The CRDs will ve updated together with the prometheus-operator version, by a scrtipt: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/hack/update_crds.sh
> 
> So please always use it and also update prometheus operator to 0.55.1, updating all docker iamges, dadhboards and rules too.

sorry for the delay, but i have done this. of note, the CRD still does not support enableRemoteWriteReceiver.

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -23,8 +23,8 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 34.9.0
-appVersion: 0.55.0
+version: 35.2.1
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]: New feature should raise minor version.
```suggestion
version: 35.3.0
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/Chart.yaml (hunk 1 above)
Comment (daveoy) [reply charts/kube-prometheus-stack/Chart.yaml]: done

**Review by monotek — COMMENTED** [review]
i've updated the crd download script to work without input, taking the version form the Chart.yaml.

**Review by monotek — APPROVED** [review]

**Comment by daveoy** [comment]
> i've updated the crd download script to work without input, taking the version form the Chart.yaml.
> 
> 

Great idea

Thank you for merging.

**Pull request merged**